### ISSUE TITLE
Establish testing of containerized apps

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/containers#4]
Could probably be as simple as regression tests.

Existing tests:
- For Flywheel gears generated by gearificator, ideally should just reuse those tests defined under `tests/` subdirectories of https://github.com/yarikoptic/gearificated-nipype/tree/master/gears/nipype
- For some boutiques -- [could come from boutiques specs](https://github.com/boutiques/boutiques#test-your-tool)

So it seems that we need a testing framework which for image(s) would list a set of "tests drivers" to pickup/execute tests, so we could point to various tests specifications and get them executed.  

- IIRC boutiques testing implementation had some nice "integration" within pytest but I've not tried it yet
- [good old testkraut](http://github.com/neurodebian/testkraut) by @mih defines tests specification (see [testcase.py](https://github.com/neurodebian/testkraut/blob/master/testkraut/testcase.py) with examples under [localtests](https://github.com/neurodebian/testkraut/tree/master/testkraut/tests/localtests) and [populates](https://github.com/neurodebian/testkraut/blob/master/testkraut/tests/test_localtests.py#L30) tests for nose (or pytest!) to pick them up -- so something to base things on possibly
<details>
<summary>NB with `pytest` it is possible to execute individual ones! (nose relies on import, fails)</summary>

```shell
(git)hopa:~exppsy/testkraut[master]git
$> python -m pytest -s -v testkraut/tests/test_localtests.py::LocalDogFoodTests::test_check_assertions                                  
================================================================= test session starts ==================================================================
platform linux2 -- Python 2.7.14+, pytest-3.10.1, py-1.7.0, pluggy-0.8.0 -- /home/yoh/deb/gits/pkg-exppsy/testkraut/venvs/dev/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/yoh/deb/gits/pkg-exppsy/testkraut/.hypothesis/examples')
rootdir: /home/yoh/deb/gits/pkg-exppsy/testkraut, inifile:
plugins: xdist-1.26.1, localserver-0.5.0, forked-1.0.1, cov-2.6.0, hypothesis-3.71.11, celery-4.2.1
collected 1 item                                                                                                                                       

testkraut/tests/test_localtests.py::LocalDogFoodTests::test_check_assertions PASSED
```

Attn @djarecka , @glatard

**Comment by glatard** [comment]
Indeed, Boutiques generates pytest tests from their JSON definition and runs them with pytest. For example, you can run `bosh test tests_good.json` where `tests_good.json` is https://raw.githubusercontent.com/boutiques/boutiques/master/tools/python/boutiques/schema/examples/tests_good.json

**Comment by yarikoptic** [comment]
Notes (to myself primarily): 
- even if tests are bundled within the container, the ultimate testing scenario is to execute them "from outside", i.e. input data (where any) should be provided via bind mounts etc, options specified/passed through command line since the primary goal is to verify correct operation of the container as to be used in a present environment.  
- Having said above, it might be nice to also be able to run "internal" tests within the container, again given the default bind mounts etc we would provide to see that there is no side effects of any kind from our way to execute the container
